lib/facter/lldb.rb

```diff
@@ -0,0 +1,14 @@
+require 'facter'
+require 'json'
+
+Facter.add(:lldp) do
+  confine do
+    Facter::Util::Resolution.which('lldpctl')
+  end
+
+  setcode do
+    data = Facter::Util::Resolution.exec('lldpctl -f json')
+    json = JSON.parse(data)
```

Comment:
I had some strange edge cases where lldpctl didn't produce any output / printend warnings before the actual json. that's why the script valides the output with jq.

Reply:
I did wonder why you used jq, could add a try; catch here.  Guessing you don't remember the edge cases or have test data to tickle the bug?

As mentioned above we do make use of this binary to [produce a similar fact](https://github.com/wikimedia/puppet/blob/production/modules/base/lib/facter/lldp.rb) and don't see any issues with.  however we use `-f xml` so depending on where the bug is might not hit it

Reply:
added a rescue